cli/command/image/build_test.go

```diff
@@ -68,3 +68,24 @@ func TestRunBuildDockerfileFromStdinWithCompress(t *testing.T) {
 	sort.Strings(actual)
 	assert.Equal(t, []string{dockerfileName, ".dockerignore", "foo"}, actual)
 }
+
+// TestRunBuildFromLocalGitHubDirNonExistingRepo tests that build contexts
+// starting with `github.com/` are special-cased, and the build command attempts
+// to clone the remote repo, however a local directory should take precedence.
+func TestRunBuildFromLocalGitHubDirNonExistingRepo(t *testing.T) {
+
+	cmd := NewBuildCommand(&command.DockerCli{})
+	cmd.SetArgs([]string{"github.com/docker/no-such-repository"})
+	cmd.SetOutput(ioutil.Discard)
+	err := cmd.Execute()
+	assert.Contains(t, err.Error(), "unable to prepare context: unable to 'git clone'")
+
+	tmpDir, err := ioutil.TempDir("", "docker-build-from-local-dir-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	os.Chdir(tmpDir)
+	os.MkdirAll(filepath.Join(tmpDir, "github.com/docker/no-such-repository"), 0777)
+	err = cmd.Execute()
+	assert.Contains(t, err.Error(), "unable to prepare context: unable to evaluate symlinks in Dockerfile path:")
```

Comment:
Since the expectation here is "does build from local directory" maybe this would be more clear by adding a simple `Dockerfile` to the directory and checking for no error?